README.rst

```diff
@@ -410,37 +613,798 @@ Please refer
 `troubleshooting issues related to geospatial libraries
 <https://docs.djangoproject.com/en/2.1/ref/contrib/gis/install/#library-environment-settings/>`_.
 
-Talks
------
+How to use configuration variables
+----------------------------------
 
-- `OpenWISP2 - a self hosted solution to control OpenWRT/LEDE devices
-  <https://fosdem.org/2017/schedule/event/openwisp2/>`_ (FOSDEM 2017)
+Sometimes the configuration is not exactly equal on all the devices,
+some parameters are unique to each device or need to be changed
+by the user.
 
-Contributing
-------------
+In these cases it is possible to use configuration variables in conjunction
+with templates, this feature is also known as *configuration context*, think of
+it like a dictionary which is passed to the function which renders the
+configuration, so that it can fill variables according to the passed context.
 
-Please read the `OpenWISP contributing guidelines
-<http://openwisp.io/docs/developer/contributing.html>`_
-and also keep in mind the following:
+The different ways in which variables are defined are described below.
 
-1. Announce your intentions in the `OpenWISP Mailing List <https://groups.google.com/d/forum/openwisp>`_
-2. Fork this repo and install it
-3. Follow `PEP8, Style Guide for Python Code`_
-4. Write code
-5. Write tests for your code
-6. Ensure all tests pass
-7. Ensure test coverage does not decrease
-8. Document your changes
-9. Send pull request
+Predefined device variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-.. _PEP8, Style Guide for Python Code: http://www.python.org/dev/peps/pep-0008/
+Each device gets the following attributes passed as configuration variables:
 
-Changelog
----------
+* ``id``
+* ``key``
+* ``name``
+* ``mac_address``
 
-See `CHANGES <https://github.com/openwisp/openwisp-controller/blob/master/CHANGES.rst>`_.
+User defined device variables
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-License
+In the device configuration section, you can access the context
+field by clicking on "Advanced Options (show)".
+
+.. image:: https://raw.githubusercontent.com/openwisp/openwisp-radius/master/docs/images/device-advanced.png
```

Comment:
:warning: openwisp-radius (fix other occurrences as well please)

Reply:
Probably, happened when copy-pasted the docs, made an error in re-writing the link. Fixed.